doc/tutorials/neat/neat.txt

```diff
@@ -0,0 +1,248 @@
+/*!
+
+@file neat.txt
+@author Rahul Ganesh Prabhu
+@brief Tutorial for how to use neat in mlpack.
+
+@page neattutorial NeuroEvolution of Augmenting Topologies tutorial
+
+@section intro_neattut Introduction
+
+NeuroEvolution of Augmenting Topologies (NEAT) is a genetic algorithm proposed by Kenneth Stanley that can evolve networks of unbound complexity from simple networks and "complexifying" through different genetic operators. It has been used to create "genetic art" and an agent that can complete the first level in Super Mario World.
+
+This implementation of NEAT also has support for Phased Searching, which is a searching scheme proposed by Colin Green, involving alternating between complexification and simplification, based on the mean complexity of the population. Here, complexity is defined as the number of connections in the genome.
+
+@code
+@article{Stanley:2002:ENN:638553.638554,
+ author = {Stanley, Kenneth O. and Miikkulainen, Risto},
+ title = {Evolving Neural Networks Through Augmenting Topologies},
+ journal = {Evol. Comput.},
+ issue_date = {Summer 2002},
+ volume = {10},
+ year = {2002}, 
+@endcode
+
+@section toc_neattut Table of Contents
+
+A list of all the sections this tutorial contains.
+
+  - \ref intro_neattut
+  - \ref toc_neattut
+  - \ref neat_neattut
+    - \ref neat_genome
+    - \ref neat_user
+  - \ref neat_template_neattut
+    - \ref neat_tasktype_neattut
+    - \ref neat_selection_neattut
+  - \ref further_doc_neattut
+
+@section neat_neattut The NEAT class
+
+@subsection neat_genome The Genome class
+
+The Genome class is the representation of a genome in the population. It contains a list of connection genes that represent it's network. The parameters of this network are easily accessed by calling the \c Parameters() function, which returns a kind of adjacency matrix, where the element at (i, j) represents the weight of a connection from node ID i to node ID j.
+
+It is important to know the scheme by which these node IDs are allotted:
+
+  - The bias node always has ID 0.
+  - The input nodes are given the IDs from 1 to inputNodeCount
+  - The output nodes are given the IDs from inputNodeCount + 1 to inputNodeCount + outputNodeCount.
+  - The hidden nodes are given IDs greater than inputNodeCount + outputNodeCount.
+
+The bias of the genome can also be found using the Bias() method.
+
+The genome is also what is used to evaluate an input based on the network it represents. This is done by using the genome's Evaluate() function:
+
+@code
+#include <mlpack/methods/neat/neat.hpp>
+
+arma::vec input = arma::randn<arma::vec>(inputNodeCount);
+
+// Assuming a Genome object called genome has already been instantiated.
+arma::vec output = genome.Evaluate(input);
```

Comment:
Do you think we should instantiate a Genome, that way a use could just copy past the lines from the example.